lib/transformer/dataset_generator/analyzer.rb

```diff
@@ -0,0 +1,21 @@
+module Transformer
+  class DatasetGenerator
+    module Analyzer
```

Comment:
For simple callables (which have only a `call` method and nothing else) you might want to consider swapping them out for a lambda:

```ruby
module Transformer
  module DatasetGenerator
    Analyzer = lambda do |cast|
      values = Caster.cast(cast)
      dataset = Atlas::Dataset::Derived.find(cast.area)

      # ...
    end
  end
end
```

Ditto for Destroyer, Scaler, and Validator.